frontend/ui/widget/radiobutton.lua

```diff
@@ -0,0 +1,164 @@
+--[[--
+Widget that shows a radiobutton checked (`◉`) or unchecked (`◯`)
+or nothing of the same size.
+
+Example:
+
+    local RadioButton = require("ui/widget/radiobutton")
+    local parent_widget = FrameContainer:new{}
+    table.insert(parent_widget, RadioButton:new{
+        checkable = false, -- shows nothing when false, defaults to true
+        checked = function() end, -- whether the box is checked
+    })
+    UIManager:show(parent_widget)
+
+]]
+
+local Blitbuffer = require("ffi/blitbuffer")
+local Device = require("device")
+local Font = require("ui/font")
+local FrameContainer = require("ui/widget/container/framecontainer")
+local Geom = require("ui/geometry")
+local GestureRange = require("ui/gesturerange")
+local InputContainer = require("ui/widget/container/inputcontainer")
+local LeftContainer = require("ui/widget/container/leftcontainer")
+local TextWidget = require("ui/widget/textwidget")
+local UIManager = require("ui/uimanager")
+
+local RadioButton = InputContainer:new{
+    checkable = true,
+    checked = false,
+    enabled = true,
+    face = Font:getFace("smallinfofont"),
+    background = Blitbuffer.COLOR_WHITE,
+    width = 0,
+    height = 0,
+}
+
+function RadioButton:init()
+    self._checked_widget = TextWidget:new{
+        text = "◉ " .. self.text,
+        face = self.face,
+    }
+    self._unchecked_widget = TextWidget:new{
+        text = "◯ " .. self.text,
+        face = self.face,
+    }
+    self._empty_widget = TextWidget:new{
+        text = "" .. self.text,
+        face = self.face,
+    }
+    self._widget_size = self._unchecked_widget:getSize()
+    if self.width == nil then
+        self.width = self._widget_size.w
+    end
+    self._radio_button = self.checkable
+                             and (self.checked and self._checked_widget or self._unchecked_widget)
+                             or self._empty_widget
+    self:_initFrame()
+    self.dimen = self.frame:getSize()
+    if Device:isTouchDevice() then
+        self.ges_events = {
+            TapCheckButton = {
+                GestureRange:new{
+                    ges = "tap",
+                    range = self.dimen,
+                },
+                doc = "Tap Button",
+            },
+            HoldCheckButton = {
+                GestureRange:new{
+                    ges = "hold",
+                    range = self.dimen,
+                },
+                doc = "Hold Button",
+            }
+        }
+    end
+end
+
+function RadioButton:_initFrame()
```

Comment:
I think most of the other widgets with this kind of 2 steps init, call this method `:update()`.
And although you don't need that as you don't have any subwidget that would really need it, you could add just there:
```lua
    if self[1] then
        self[1]:free()
    end
```

Reply:
True, might as well do that in case there's a BB or RichTextWidget in there in the future or something.